packages/vkui/src/lib/children.ts

```diff
@@ -0,0 +1,42 @@
+// copied from react-children-utilities
+// @see https://github.com/fernandopasik/react-children-utilities/tree/main
+
+import * as React from 'react';
+import { Children, isValidElement } from 'react';
+
+export const hasChildren = (
+  element: React.ReactNode,
+): element is React.ReactElement<{ children: React.ReactNode | React.ReactNode[] }> =>
+  isValidElement<{ children?: React.ReactNode[] }>(element) && Boolean(element.props.children);
+
+export const childToString = (child?: React.ReactNode): string => {
+  if (typeof child === 'undefined' || child === null || typeof child === 'boolean') {
+    return '';
+  }
+
+  if (JSON.stringify(child) === '{}') {
+    return '';
+  }
+
+  return (child as number | string).toString();
+};
+
+export const getTextFromChildren = (children: React.ReactNode | React.ReactNode[]): string => {
+  if (!(children instanceof Array) && !isValidElement(children)) {
+    return childToString(children);
+  }
+
+  return Children.toArray(children).reduce((text: string, child: React.ReactNode): string => {
+    let newText = '';
+
+    if (isValidElement(child) && hasChildren(child)) {
+      newText = getTextFromChildren(child.props.children);
+    } else if (isValidElement(child) && !hasChildren(child)) {
+      newText = '';
+    } else {
+      newText = childToString(child);
+    }
```

Comment:
я ещё у них заметил, что предикаты можно вынести в переменные

```suggestion
    const isValidElementResult = isValidElement<{ children?: React.ReactNode | React.ReactNode[] }>(child);
    const hasChildren = isValidElementResult && 'children' in child.props;
    
    if (isValidElementResult && hasChildren) {
      newText = getTextFromChildren(child.props.children);
    } else if (isValidElementResult && !hasChildren) {
      newText = '';
    } else {
      newText = childToString(child);
    }
```

а то получается при вхождениях, когда нет `child.props.children`, ребята по второму разу проверяют одно и тоже, получается по два раза три функции срабатывали `isValideElement(child) && isValideElement(child) && Boolean(child.props.children)`

Reply:
ну и получается функция `hasChildren` нам пока не нужна